File: src/MaterialAir.cpp
// this file is covered by the General Public License version 2 or later
// please see GPL.html for more details and licensing issues
// copyright Etienne de Foras ( the author )  mailto: etienne.deforas@gmail.com

#include "MaterialAir.h"

//////////////////////////////////////////////////////////////////////////////
MaterialAir::MaterialAir()
{
    _sName="Air";
    _sFormula="Custom";
    set_maker("internal");
    _iSolidColor=0xffffff;
}
//////////////////////////////////////////////////////////////////////////////
MaterialAir::MaterialAir(const MaterialAir& m):  Glass(m)
{ }
//////////////////////////////////////////////////////////////////////////////
Glass* MaterialAir::clone() const
{
    return new MaterialAir(*this);
}
//////////////////////////////////////////////////////////////////////////////
double MaterialAir::compute_index(double dLambdaMicrons)
{
    double sigma2=1./(dLambdaMicrons*dLambdaMicrons); //dLambda in microns
    return 1. + 6.4328e-5 + 2.94981e-2/(146.-sigma2) + 2.554e-4/(41.- sigma2);

    //source : http://olivier.fournet.free.fr/science_et_physique/refraction_air.htm
    //  with 0.03% de CO2 , 101325 Pa , T= 288.15 K
}
//////////////////////////////////////////////////////////////////////////////

File: python/MaterialAir.py
# this file is covered by the  GNU LESSER GENERAL PUBLIC LICENSE Version 3 or later
# please see LICENSE.txt for more details and licensing issues
# copyright Etienne de Foras ( the author )  mailto: etienne.deforas@gmail.com

import Glass

class MaterialAir(Glass):
    def __init__(self):
        self._sName = "Air"
        self._sFormula = "Custom"
        self.set_maker("internal")
        self._iSolidColor = 0xffffff

    def clone(self):
        return MaterialAir()

    def compute_index(self, dLambdaMicrons):
        sigma_squared = 1.0 / (dLambdaMicrons * dLambdaMicrons)  # dLambda in microns
        return 1.0 + 6.4328e-5 + 2.94981e-2 / (146.0 - sigma_squared) + 2.554e-4 / (41.0 - sigma_squared)

        # source : http://olivier.fournet.free.fr/science_et_physique/refraction_air.htm
        #  with 0.03% de CO2 , 101325 Pa , T= 288.15 K